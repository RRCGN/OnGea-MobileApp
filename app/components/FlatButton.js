/**
 * Flat Button (no background) with touch feedback
 * @flow
 */

import React from 'react'
import {
  View,
  TouchableNativeFeedback,
  TouchableOpacity,
  Platform,
  StyleSheet
} from 'react-native'


type Props = {
  onPress: () => void,
  children: ReactElement<*>
}

const FlatButton = ({ onPress, children }: Props) => {
  // Decide: TouchableOpacity for iOS or TouchableNativeFeedback for Android
  const Touch =
    Platform.OS === 'ios' ? TouchableOpacity : TouchableNativeFeedback

  // Use SelectableBackground only under Android
  // ...as callback, so iOS doesn't get mad.
  const backgroundProps = Platform.select({
    ios: {},
    android: {
      background: () => (TouchableNativeFeedback.SelectableBackground())
    }
  })

  return (
    <Touch
      onPress={onPress}>
      <View style={styles.container}>
        {children}
      </View>
    </Touch>
  )
}

export default FlatButton


const styles = StyleSheet.create({
  container: {
    marginRight: 8,
    padding: 8,
    borderRadius: 2
  }
})
