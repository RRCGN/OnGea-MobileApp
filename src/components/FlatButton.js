import React from 'react'
import PropTypes from 'prop-types'
import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
  ActivityIndicator
} from 'react-native'

const FlatButton = props => {
  return (
    <TouchableOpacity
      style={props.style}
      disabled={props.disabled}
      onPress={props.onPress}
    >
      <View style={styles.base}>
        {props.isLoading ? (
          <ActivityIndicator size="small" color="#6c6c6c" />
        ) : (
          <Text style={styles.text}>{props.children}</Text>
        )}
      </View>
    </TouchableOpacity>
  )
}

FlatButton.propTypes = {
  onPress: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
  style: PropTypes.any,
  loading: PropTypes.bool
}

export default FlatButton

const styles = StyleSheet.create({
  base: {
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 6,
    backgroundColor: '#eaeaea',
    paddingVertical: 14,
    paddingHorizontal: 24
  },
  text: {
    fontSize: 14,
    color: '#6c6c6c'
  }
})
