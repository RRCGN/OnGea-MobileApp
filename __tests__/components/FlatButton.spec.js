import { View } from 'react-native'
import React from 'react'
import FlatButton from '../../app/components/FlatButton'

import renderer from 'react-test-renderer'

describe('<FlatButton />', () => {
  test('renders correctly with custom styles', () => {
    const tree = renderer.create(
      <FlatButton onPress={() => {}} style={{ backgroundColor: 'red' }} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('renders correctly with label', () => {
    const tree = renderer.create(
      <FlatButton onPress={() => {}} label="Foo" />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('renders correctly with label and color', () => {
    const tree = renderer.create(
      <FlatButton onPress={() => {}} label="Foo" color="red"  />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('renders correctly with icon', () => {
    const tree = renderer.create(
      <FlatButton onPress={() => {}} icon={<View />} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
