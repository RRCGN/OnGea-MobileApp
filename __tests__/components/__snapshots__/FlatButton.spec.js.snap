exports[`<FlatButton /> renders correctly with custom styles 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}>
  <View>
    <View
      style={
        Array [
          Object {
            "borderRadius": 2,
            "padding": 8,
          },
          Object {
            "backgroundColor": "red",
          },
        ]
      } />
  </View>
</View>
`;

exports[`<FlatButton /> renders correctly with icon 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}>
  <View>
    <View
      style={
        Array [
          Object {
            "borderRadius": 2,
            "padding": 8,
          },
          undefined,
        ]
      }>
      <View />
    </View>
  </View>
</View>
`;

exports[`<FlatButton /> renders correctly with label 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}>
  <View>
    <View
      style={
        Array [
          Object {
            "borderRadius": 2,
            "padding": 8,
          },
          undefined,
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
            },
            Object {
              "color": "#2196F3",
              "fontWeight": "bold",
              "textAlign": "center",
            },
          ]
        }>
        FOO
      </Text>
    </View>
  </View>
</View>
`;

exports[`<FlatButton /> renders correctly with label and color 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}>
  <View>
    <View
      style={
        Array [
          Object {
            "borderRadius": 2,
            "padding": 8,
          },
          undefined,
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "red",
            },
            Object {
              "color": "#2196F3",
              "fontWeight": "bold",
              "textAlign": "center",
            },
          ]
        }>
        FOO
      </Text>
    </View>
  </View>
</View>
`;
